src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs

```diff
@@ -372,4 +381,39 @@ static bool TryFetchException(object payload, out Exception exc)
             return true;
         }
     }
+
+    private static string GetErrorType(Exception exc)
+    {
+#if NET8_0_OR_GREATER
+        // For net8.0 and above exception type can be found using HttpRequestError.
+        // https://learn.microsoft.com/dotnet/api/system.net.http.httprequesterror?view=net-8.0
+        var httpRequestException = exc as HttpRequestException;
+        if (httpRequestException != null)
+        {
+            return httpRequestException.HttpRequestError switch
+            {
+                HttpRequestError.NameResolutionError => "name_resolution_error",
+                HttpRequestError.ConnectionError => "connection_error",
+                HttpRequestError.SecureConnectionError => "secure_connection_error",
+                HttpRequestError.HttpProtocolError => "http_protocol_error",
+                HttpRequestError.ExtendedConnectNotSupported => "extended_connect_not_supported",
+                HttpRequestError.VersionNegotiationError => "version_negotiation_error",
+                HttpRequestError.UserAuthenticationError => "user_authentication_error",
+                HttpRequestError.ProxyTunnelError => "proxy_tunnel_error",
+                HttpRequestError.InvalidResponse => "invalid_response",
+                HttpRequestError.ResponseEnded => "response_ended",
+                HttpRequestError.ConfigurationLimitExceeded => "configuration_limit_exceeded",
+
+                // Fall back to the exception type name in case of HttpRequestError.Unknown
+                _ => exc.GetType().FullName,
+            };
+        }
+        else
+        {
+            return exc.GetType().FullName;
+        }
```

Comment:
nit: If you tweak the preprocessors a bit you could remove this block and just let the method flow to the final return.